### DISCUSSION TITLE
Is https://github.com.cnpmjs.org/xianyi/OpenBLAS/archive/v0.3.9.tar.gz error?

### DISCUSSION THREAD

**Discussion by yy9991818** [milvus-io/milvus#15756] [Q&A and General discussion] [answered]
I cloned milvus by  git clone https://github.com/milvus-io/milvus.git
when I compile it by make , one problem occurs,  OpenBLAS cannot download by https://github.com.cnpmjs.org/xianyi/OpenBLAS/archive/v0.3.9.tar.gz, I put it in browser, it cannot access either,
Is this url right? 

I can get it by https://codeload.github.com/xianyi/OpenBLAS/tar.gz/refs/tags/v0.3.9

**Comment by yhmo** [comment, marked as answer]
The original address is https://github.com/xianyi/OpenBLAS/archive/v0.3.9.tar.gz
It was redirected to https://github.com.cnpmjs.org/xianyi/OpenBLAS/archive/v0.3.9.tar.gz for unknown reason.

**Reply by yy9991818** [reply]
In ${MILVUS_SOURCE_HOME}/cmake_build/3rdparty_download/knowhere-subbuild/src/knowhere_ep/cmake/ThirdPartyPackagesKw.cmake
when change https://github.com.cnpmjs.org/xianyi/OpenBLAS/archive/v0.3.9.tar.gz  to https://github.com/xianyi/OpenBLAS/archive/v0.3.9.tar.gz, it works fine.

**Comment by matrixji** [comment]
For CMake, it supports multiple URLs since 3.7, seems it is better to set multiple URLs for the external resources.
Maybe the first one use the releases url from GitHub, and some others with mirrors url.
